design/token.go

```diff
@@ -5,10 +5,49 @@ import (
 	a "github.com/goadesign/goa/design/apidsl"
 )
 
+// externalToken represents a token object
+var externalToken = a.MediaType("application/vnd.externalToken+json", func() {
+	a.TypeName("ExternalToken")
+	a.Description("External Provider Token")
+	a.Attributes(func() {
+		a.Attribute("access_token", d.String, "The token associated with the identity for the specific external provider")
+		a.Attribute("scope", d.String, "The scope associated with the token")
+		a.Attribute("token_type", d.String, "The type of the toke, example : bearer")
+		a.Required("access_token", "scope", "token_type")
+	})
+
+	a.View("default", func() {
+		a.Attribute("access_token")
+		a.Attribute("scope")
+		a.Attribute("token_type")
+		a.Required("access_token", "scope", "token_type")
+	})
+
+})
+
 var _ = a.Resource("token", func() {
 
 	a.BasePath("/token")
 
+	a.Action("Retrieve", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET(""),
+		)
+		a.Params(func() {
+			a.Param("for", d.String, "The resource for which the external provider token is being fetched")
+			a.Param("scope", d.String, "The scope for which the token is being fetched") // #428
+			a.Required("for")
+		})
+		a.Description("Get the external provider token")
+		a.Response(d.OK, externalToken)
+		a.Response(d.NotModified)
```

Comment:
Was thinking of adding it initially since the token would hardly change -
 should we support caching? Or maybe after we moved over everything.

Turned it off for now e54df1c